engine/execution/computation/computer/computer.go

```diff
@@ -95,111 +104,156 @@ func (e *blockComputer) ExecuteBlock(
+		// execute the first collection
+		colView = stateView.NewChild()
+		txIndex, err1 = e.executeCollection(blockSpan, txIndex, blockCtx, colView, programs, collections[0], res)
+		if err1 != nil {
+			return nil, fmt.Errorf("failed to execute collection: %w", err1)
+		}
+		prevColView = colView
+		err1 = stateView.MergeView(colView)
+		if err1 != nil {
+			return nil, fmt.Errorf("cannot merge view: %w", err1)
 		}
 
-		gasUsed += gas
-
-		txIndex = nextIndex
-		events = append(events, collEvents...)
-		serviceEvents = append(serviceEvents, collServiceEvents...)
-		blockTxResults = append(blockTxResults, txResults...)
-
-		interactions[i] = collectionView.(*delta.View).Interactions()
+		// executing collections
+		for _, collection := range collections[1:] {
+			wg.Add(2)
+			colView = stateView.NewChild()
+
+			go func() {
+				txIndex, err1 = e.executeCollection(blockSpan, txIndex, blockCtx, colView, programs, collection, res)
+				wg.Done()
+			}()
+
+			go func() {
+				stateCommit, err2 = e.commitView(blockSpan, prevColView, stateCommit, res)
+				wg.Done()
+			}()
+
+			wg.Wait()
+			if err1 != nil {
+				return nil, fmt.Errorf("failed to execute collection: %w", err1)
+			}
+			if err2 != nil {
+				return nil, fmt.Errorf("failed to commit view while executing a collection: %w", err2)
+			}
+			prevColView = colView
+			err = stateView.MergeView(colView)
+			if err != nil {
+				return nil, fmt.Errorf("cannot merge view: %w", err)
+			}
+		}
 
-		err = stateView.MergeView(collectionView)
-		if err != nil {
-			return nil, fmt.Errorf("cannot merge view: %w", err)
+		// commit last view
+		stateCommit, err2 = e.commitView(blockSpan, prevColView, stateCommit, res)
+		if err2 != nil {
+			return nil, fmt.Errorf("failed to commit view while executing a collection: %w", err2)
 		}
 	}
 
-	// system chunk
-	systemChunkView := stateView.NewChild()
+	// TODO make this part also parallel (right now it doesn't worth the time)
+	// executing system chunk
 	e.log.Debug().Hex("block_id", logging.Entity(block)).Msg("executing system chunk")
+	collectionView := stateView.NewChild()
+	_, err = e.executeSystemCollection(blockSpan, txIndex, collectionView, programs, res)
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute system chunk transaction: %w", err)
+	}
+	stateCommit, err = e.commitView(blockSpan, collectionView, stateCommit, res)
+	err = stateView.MergeView(collectionView)
+	if err != nil {
+		return nil, fmt.Errorf("cannot merge view: %w", err)
+	}
+	res.StateReads = stateView.(*delta.View).ReadsCount()
+	return res, nil
+}
 
-	var colSpan opentracing.Span
-	colSpan, _ = e.tracer.StartSpanFromContext(ctx, trace.EXEComputeSystemCollection)
+func (e *blockComputer) commitView(
+	blockSpan opentracing.Span,
+	collectionView state.View,
+	state flow.StateCommitment,
+	res *execution.ComputationResult,
+) (flow.StateCommitment, error) {
+	span := e.tracer.StartSpanFromParent(blockSpan, trace.EXECommitDelta)
```

Comment:
We run this method in goroutines, not sure if it won't mess up tracking without synchronization here

Reply:
it doesn't cause any issue, trace module is threadsafe and we can spin off several child spans in parallel from the same parent.